src/controllers/user/user.controller.js

```diff
@@ -0,0 +1,126 @@
+const userAccountService = require("../../services/user/user.service");
+const passwordUtils = require("../../utils/password.utils");
+const crypto = require("crypto");
+const jwt = require("jsonwebtoken");
+const { validationResult } = require("express-validator");
+const { JWT_SECRET_KEY } = require("../../configs/env.config");
+
+const addUserAccount = async (req, res) => {
+  //validating the request body
+  const errors = validationResult(req);
```

Comment:
Your request will never reach this controller if there are any validation errors _(since you're validating the requests in one of your middleware)_. Hence, this errors checking is redundant in the controllers